docs/embedded/flexpret.mdx

```diff
@@ -0,0 +1,179 @@
+---
+title: FlexPRET
+description: Developing LF Programs for FlexPRET.
+---
+# Overview
+
+FlexPRET is a precision-timed (PRET) machine designed for mixed-criticality systems. As of 2024, PRET machines are an open field of research. Refer to its [github page](https://github.com/pretis/flexpret) for more in-depth information. FlexPRET either needs to be emulated or run on a Field-Programmable Gate Array  (FPGA). In this guide we will show you how to pair FlexPRET with Lingua Franca, leaving you with precise software execution.
+
+As a developer, you should be aware of a significant difference between FlexPRET's notion of threads versus other platforms. FlexPRET uses fined-grained *hardware* threads, as opposed to the usual (software) threads. This means that if multiple threads are running, they can swap every clock cycle. In addition, hardware threads are designed to be isolated, meaning the execution time of one thread will not affect the execution time of another, as opposed to concurrent software threads. Since the threads are implemented in hardware, FlexPRET can only have a maximum of eight.
+
+# Getting started
+
+## Development environment setup
+
+1. We start out by setting up a development environment. Clone the 
+`lf-flexpret-template` from Github:
+```
+git clone --recurse-submodules https://github.com/lf-lang/lf-flexpret-template lf-flexpret-workspace && cd lf-flexpret-workspace
+```
+2. Source the `env.bash` or `env.fish` to set up necessary environment variables. You will need to do this every time, so it could be a good idea to add this to your `~/.bashrc`.
+```
+source env.bash
+```
+```
+source env.fish
+```
+
+## FlexPRET build
+
+3. Now we will build the FlexPRET emulator. Step into the FlexPRET directory and build the default configuration with `cmake`.
+```
+cd $FP_PATH
+cmake -B build && cd build && make
+```
+
+(It is possible to name the build folder something other than `build`, but `bin` is reserved.)
+
+4. You should now install the FlexPRET build to the FlexPRET software development kit (SDK). This provides the SDK with the necessary knowledge of FlexPRET's hardware configuration, such as the amount of data memory available.
```

Comment:
Could you rephrase "install the FlexPRET build to the FlexPRET software development kit"

Reply:
Ok, I think it's better now :)